### ISSUE TITLE
Document downloading source packages

### ISSUE THREAD

**Issue by khbecker** [canonical/real-time-ubuntu-docs#48]
We should document how Pro users can pull down the source for linux-realtime.

First they need to have realtime-kernel enabled with pro (or at least get access with `sudo pro enable realtime-kernel --access-only`) and then run `sudo apt-get source <realtime linux package name>`. For example, for the latest version currently, you would run `sudo apt-get source linux-image-unsigned-6.8.1-1003-realtime`. This will download the kernel tree for that version in a folder labelled linux-realtime-6.8.1.

**Comment by marcelocerri** [comment]
Some kernels don't have have linux-image-unsigned (because they are not signed). A more generic solution would be to use the modules package instead.

**Comment by khbecker** [comment]
Thanks for the info. Andy let me know of another way to do it that is more generic: `sudo apt-get source --only-source linux-realtime`

**Comment by benjamin051000** [comment]
> Thanks for the info. Andy let me know of another way to do it that is more generic: `sudo apt-get source --only-source linux-realtime`

Hm, that didn't work for me:

```
❯ sudo apt-get source --only-source linux-realtime
Reading package lists... Done
E: You must put some 'deb-src' URIs in your sources.list
```
Do I need to reboot after `sudo pro enable realtime-kernel --access-only`, even though that command doesn't actually swap out the kernel?

**Comment by benjamin051000** [comment]
https://askubuntu.com/questions/1512042/ubuntu-24-04-getting-error-you-must-put-some-deb-src-uris-in-your-sources-list this shows that I may need to add `deb-src` to the repo sources in the `ubuntu.sources` file.

**Comment by khbecker** [comment]
Yes that's correct, you also need to add `deb-src` to your sources file.

**Comment by benjamin051000** [comment]
@khbecker It still doesn't appear to work... see above. I'm on noble but I do have a (free) pro account... so I _should_ be able to download it, right?

**Comment by khbecker** [comment]
Yes, I think you should be able to. I'll have to set up a clean VM and try again.

**Comment by khbecker** [comment]
I was able to retest. Give this a try.

For LTS releases (jammy/noble) you need to configure a few things: 
- attach your pro token
- enable access to realtime:`sudo pro enable realtime-kernel --access-only`
- enable `deb-src` in `/etc/apt/sources.list.d/ubuntu-realtime-kernel.sources`
- run `sudo apt update`

For interim releases (oracular/plucky), you need to enable `deb-src` to `/etc/apt/sources.list.d/ubuntu.sources`.

Then for either interim releases or LTS releases, you need to install `dpkg-dev`.

Then to get the sources, run `sudo apt-get source --only-source linux-realtime`. This will download the source packages and extract them. If you're in a VM, make sure you have enough disk space. The download of the packages and the extraction of those files takes almost 2GB of space on their own.

**Comment by benjamin051000** [comment]
Ah, I see. I put `deb-src` into the wrong file. I just put it in `ubuntu-realtime-kernel.sources` and now the `apt-get` command appears to work!

Thanks for investigating this @khbecker. I'll be sure to include two paths in the instructions depending on which distro you're running.

**Comment by benjamin051000** [comment]
Are there any major differences if we install the RT snap?

**Comment by farshidtz** [comment]
@benjamin051000 The RT snap is meant for Ubuntu Core, which doesn't support debian packages. I believe the documentation expected here is only for classic Ubuntu, i.e. Ubuntu Server and Ubuntu Desktop.
